src/catchup/ApplyBucketsWork.cpp

```diff
@@ -145,6 +149,35 @@ ApplyBucketsWork::startLevel()
 BasicWork::State
 ApplyBucketsWork::onRun()
 {
+    if (mResolveMerges && mDelayTimer)
+    {
+        CLOG(INFO, "History") << "ApplyBucketsWork: application completed; "
+                                 "waiting for merge resolution";
+        auto& bl = mApp.getBucketManager().getBucketList();
+        bl.resolveAnyReadyFutures();
+        if (bl.futuresAllResolved())
+        {
+            return State::WORK_SUCCESS;
+        }
+        else
+        {
+            std::weak_ptr<ApplyBucketsWork> weak(
+                std::static_pointer_cast<ApplyBucketsWork>(shared_from_this()));
+            auto handler = [weak](asio::error_code const& ec) {
+                auto self = weak.lock();
+                if (self)
+                {
+                    self->wakeUp();
+                }
+            };
+
+            // Check back later
+            mDelayTimer->expires_from_now(std::chrono::milliseconds(500));
```

Comment:
This seems like it'll spin twice a second for .. potentially as long as the merge runs? Current merges are short-ish but they can potentially run for minutes, hours..

Reply:
The longest ones we've seen after shadow removal were under a minute. With the current state of the system, something like 0.5-1 s seems reasonable to me, although it can be tweaked in the future. What do you think?

Reply:
*Shrug* ok. As you say, easy to change later.